src/WebSocketConnection.ts

```diff
@@ -433,7 +439,7 @@ class WebSocketConnection {
     );
   };
 
-  protected handleSocketError = (err: Error) => {
+  protected handleSocketError = (err: any) => {
```

Comment:
Because the cause can either be an Error or an event. It doesn't rlly matter what this value is, as it's just set as the cause of the thrown error

Reply:
Using `any` type tends to be pretty hacky. Generally if the type isn't well defined and it doesn't really matter then `unknown` can be used instead of `any`. 

While `any` can be anything and assigned to anything. `unknown` could be anything but can't be assigned to anything else. So it's safer to use. https://stackoverflow.com/questions/51439843/unknown-vs-any